portfolio/src/main/webapp/script.js

```diff
@@ -30,8 +30,10 @@ function addRandomGreeting() {
 async function fetchCall() {
   const response = await fetch('/hello'); 
 
-  const responseText = await response.text(); 
+  const responsejson = await response.json(); 
+  
+  var index = Math.floor(Math.random() * 3);  
```

Comment:
You should count the number of keys in `responsejson` and use that to scale the `index` to make sure that changes later in the Java code won't make line 38 have an error. 

Also: `index` can be const.

Reply:
Done.